src/Process/PointCloudProcessing.js

```diff
@@ -137,87 +146,95 @@ export default {
             }
         }
         if (commonAncestorName) {
-            return [layer.root.findChildrenByName(commonAncestorName)];
+            context.fastUpdateHint = commonAncestorName;
         }
 
         // Start updating from hierarchy root
         return [layer.root];
     },
 
     update(context, layer, elt) {
-        elt.visible = false;
-
         if (layer.octreeDepthLimit >= 0 && layer.octreeDepthLimit < elt.name.length) {
             markForDeletion(elt);
             return;
         }
 
         // pick the best bounding box
         const bbox = (elt.tightbbox ? elt.tightbbox : elt.bbox);
-        elt.visible = context.camera.isBox3Visible(bbox, layer.object3d.matrixWorld);
-        if (!elt.visible) {
-            markForDeletion(elt);
-            return;
-        }
 
-        elt.notVisibleSince = undefined;
+        if (context.fastUpdateHint && !elt.name.startsWith(context.fastUpdateHint)) {
+            if (!elt.visible) {
+                return;
+            }
+            updateMinMaxDistance(context, bbox);
+        } else {
+            elt.visible = context.camera.isBox3Visible(bbox, layer.object3d.matrixWorld);
 
-        // only load geometry if this elements has points
-        if (elt.numPoints > 0) {
-            if (elt.obj) {
-                if (elt.obj.material.update) {
-                    elt.obj.material.update(layer.material);
-                } else {
-                    elt.obj.material.copy(layer.material);
-                }
-                if (__DEBUG__) {
-                    if (layer.bboxes.visible) {
-                        if (!elt.obj.boxHelper) {
-                            initBoundingBox(elt, layer);
-                        }
-                        elt.obj.boxHelper.visible = true;
-                        elt.obj.boxHelper.material.color.r = 1 - elt.sse;
-                        elt.obj.boxHelper.material.color.g = elt.sse;
+            if (!elt.visible) {
+                markForDeletion(elt);
+                return;
+            }
+
+            const distance = updateMinMaxDistance(context, bbox);
```

Comment:
I would do a function that calculates the distance, and then with the distance, I'll refresh the context

Reply:
Err.. I didn't get this comment either. Can you reformulate?

Reply:
this is important for now I will come back to it later

Reply:
> this is important for now I will come back to it later

:confused:

Reply:
this is not important for now I will come back to it later